controllers/gitrepository_controller.go

```diff
@@ -503,8 +473,15 @@ func (r *GitRepositoryReconciler) reconcileSource(ctx context.Context,
 		optimizedClone = true
 	}
 
-	c, err := r.gitCheckout(ctx, obj, repositoryURL, authOpts, dir, optimizedClone)
+	c, err := r.gitCheckout(ctx, obj, authOpts, dir, optimizedClone)
 	if err != nil {
+		// If the error is a Stalling error, make sure to record and return that, because returning a Generic
+		// error, would force us to lose context.
+		var stalling *serror.Stalling
```

Comment:
@darkowlzz could you PTAL here? thanks

Reply:
https://github.com/fluxcd/source-controller/pull/736 should take care of this.
Thanks.